CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+spent reading your code than the time you spent writing it.
+The quicker your changes are merged to master, the less time you will need to spend rebasing and
+otherwise trying to keep up with the master branch.
+
+When fixing a bug, please start by adding a failing test that reproduces the issue.
+Create a first commit containing that test without the fix: this makes it easy to verify that the
+test correctly failed. You can then fix the bug in additional commits.
+
+When adding a new feature, thought must be given to the long term technical debt and maintenance
+that feature may require after inclusion. Before proposing a new feature that will require
+maintenance, please consider if you are willing to maintain it (including bug fixing).
+
+When addressing pull request comments, we recommend using [fixup commits](https://robots.thoughtbot.com/autosquashing-git-commits).
+The reason for this is two fold: it makes it easier for the reviewer to see what changes have been
+made between versions (since Github doesn't easily show prior versions) and it makes it easier on
+the PR author as they can set it to auto-squash the fixup commits on rebase.
+
+It's recommended to take great care in writing tests and ensuring the entire test suite has a
+stable successful outcome; eclair uses continuous integration techniques and having a stable build
+helps the reviewers with their job.
+
+We don't have hard rules around code style, but we do avoid having too many conflicting styles;
+look around and make sure you code fits well with the rest of the codebase.
+
+### Signed Commits
+
+We ask contributors to sign their commits.
+You can find setup instructions [here](https://help.github.com/en/github/authenticating-to-github/signing-commits).
+
+### Commit Message
+
+Eclair keeps a clean commit history on the master branch with well-formed commit messages.
+
+Here is a model Git commit message:
+
+```text
+Short (50 chars or less) summary of changes
+
+More detailed explanatory text, if necessary. Wrap it to about 72
+characters or so. In some contexts, the first line is treated as the
+subject of an email and the rest of the text as the body. The blank
+line separating the summary from the body is critical (unless you omit
+the body entirely); tools like rebase can get confused if you run the
+two together.
+
+Write your commit message in the present tense: "Fix bug" and not
+"Fixed bug". This convention matches up with commit messages generated
+by commands like git merge and git revert.
+
+Further paragraphs come after blank lines.
+
+- Bullet points are okay, too
+- Typically a hyphen or asterisk is used for the bullet, preceded by a
+  single space, with blank lines in between, but conventions vary here
+- Use a hanging indent
+```
+
+### Dependencies
+
+We try to minimize our dependencies (libraries and tools). Introducing new dependencies increases
+package size, attack surface and cognitive overhead.
+
```

Comment:
We should specify that eclair also runs on Android which is still using Java 7, so new dependencies must also be compatible with Java 7.

Reply:
Good point!